### PR TITLE
Extract Ktor client factory into core data

### LINKED ISSUE #374: [Refactor] Extract app-independent Ktor client configuration into :core:data
## Current state

Network client setup is split awkwardly between modules:

- `NetworkModule` lives in `:app` and provides Ktor `HttpClient` bindings.
- `:core:data` already owns shared remote request/data-source helpers, but it does not own the reusable Ktor client configuration.
- `:app`-specific concerns are mixed into `NetworkModule`, including `BuildConfig.API_BASE_URL`, `BuildConfig.DEBUG`, Hilt bindings, and auth interceptor composition.
- `:feature:auth` owns `@UnauthenticatedHttpClient`, while `:app` currently provides the matching client binding.

The old issue #308 proposed moving `NetworkModule` wholesale into `:core:data`, but that is too broad for the current module graph because `NetworkModule` depends on app and feature-specific wiring.

## Desired state

Extract only the app-independent Ktor client configuration/factory into `:core:data`.

Keep the following outside `:core:data`:

- app `BuildConfig` reads
- Hilt module bindings installed in the app graph
- authenticated vs unauthenticated client composition
- `AuthInterceptor` installation
- feature-owned qualifiers unless they are intentionally promoted to shared data infrastructure

`:app` should continue to own final dependency graph assembly, while `:core:data` owns reusable network construction primitives.

## Motivation

This keeps network infrastructure reusable without introducing invalid dependencies from `:core:data` to `:app` or `:feature:auth`.

It also aligns the implementation with ADR-002: `:core:data` should contain layer-level data infrastructure extracted from existing data responsibilities, while feature/app-specific wiring remains at the appropriate boundary.

## Affected layer

Data

## Estimated effort

M

## Suggested priority

medium

## Behavior impact

- [x] This refactoring changes no externally observable behavior

## Affected files or modules

- `core/data/src/main/kotlin/timur/gilfanov/messenger/data/...`
- `core/data/build.gradle.kts`
- `app/src/main/java/timur/gilfanov/messenger/di/NetworkModule.kt`
- `app/build.gradle.kts`
- `feature/auth/src/main/kotlin/timur/gilfanov/messenger/auth/di/UnauthenticatedHttpClient.kt` only if the qualifier is intentionally promoted to shared data infrastructure

## Related

Replaces stale issue #308.

### PR DESCRIPTION
**Pull request by timurgilfanov**
Closes #374

## Summary
Extracts reusable Ktor HTTP client construction into `:core:data` while keeping app-specific dependency graph assembly in `:app`. `:core:data` owns only app-independent network construction primitives; final graph assembly, BuildConfig reads, Hilt bindings, authenticated/unauthenticated client composition, and AuthInterceptor installation remain outside it.

## Changes
- Added a `:core:data` Ktor client factory for JSON, CIO engine, shared client configuration, and content-negotiation-only clients.
- Updated `NetworkModule` to delegate reusable client construction to `:core:data`.
- Moved production Ktor CIO/logging/content-negotiation dependencies from `:app` to `:core:data`, leaving app test dependencies where existing tests need them.
- Removed the stale app-local `KtorClient`.

## Testing
Automated formatting/static analysis, Android test Kotlin compilation, app mock-debug unit tests, and pre-commit checks passed.
- [ ] Unit tests added or updated
- [ ] Manually tested
- [x] Edge cases considered

## Screenshots / Demo
Not applicable; this PR does not affect UI.

## Acceptance criteria
- [x] This refactoring changes no externally observable behavior.

## Breaking Changes
None.

## Follow-ups
None.

### CODE REVIEW

File: app/src/main/java/timur/gilfanov/messenger/di/NetworkModule.kt
Comment (timurgilfanov) [inline app/src/main/java/timur/gilfanov/messenger/di/NetworkModule.kt:1]: Do we need a `NetworkModule` as a thin binding for DI to `KtorHttpClientFactory` in `:core:data` module? Can we have a DI module in `:core:data` instead, or `KtorHttpClientFactory` in `:app`?

File: app/build.gradle.kts [hunk 1]
```diff
@@ -334,7 +330,9 @@ dependencies {
     // Test Utilities
     testImplementation(libs.hilt.android.testing)
     testImplementation(libs.androidx.room.testing)
+    testImplementation(libs.ktor.client.content.negotiation)
     testImplementation(libs.ktor.client.mock)
+    testImplementation(libs.ktor.serialization.kotlinx.json)
```
Comment (timurgilfanov) [inline app/build.gradle.kts:335]: Why do we need new dependencies for tests?

File: app/src/main/java/timur/gilfanov/messenger/di/NetworkModule.kt
Comment (timurgilfanov) [reply app/src/main/java/timur/gilfanov/messenger/di/NetworkModule.kt:1]: Moving this DI module into `:core:data` would pull Hilt dependency and app/feature concerns like `AuthInterceptor` into `:core:data` that must be clean from this. Moving `KtorHttpClientFactory` back into `:app` would lose the extracted app-independent primitive the issue is targeting.

File: app/build.gradle.kts (hunk 1 above)
Comment (timurgilfanov) [reply app/build.gradle.kts:335]: These are not new test capabilities; they are the same Ktor content-negotiation/serialization dependencies that app unit tests already use directly when building mock `HttpClient`s. The refactor moved those dependencies out of app runtime `implementation` because production client construction now lives in `:core:data`, but the app JVM tests still import `ContentNegotiation` and Ktor’s `json` extension. Keeping them as `testImplementation` preserves test compilation without leaking those Ktor modules back into app runtime.
